### ISSUE TITLE
Spurious unused-import warning when imported variable is used as metaclass or as function annotation

### ISSUE THREAD

**Issue by pylint-bot** [pylint-dev/astroid#14]
Originally reported by: **Antony Lee (BitBucket: [anntzer](http://bitbucket.org/anntzer), GitHub: @anntzer?)**

---

```
#!python
from abc import ABCMeta

class C(metaclass=ABCMeta):
    pass

def f(arg: ABCMeta):
    pass
```

Pylint doesn't detect either use of ABCMeta and warns that the import is unused.

---
- Bitbucket: https://bitbucket.org/logilab/astroid/issue/14
